src/_pytest/runner.py

```diff
@@ -309,12 +306,9 @@ def _callfinalizers(self, colitem):
 
     def _teardown_with_finalization(self, colitem):
         self._callfinalizers(colitem)
-        if hasattr(colitem, "teardown"):
-            colitem.teardown()
+        colitem.teardown()
```

Comment:
the checks in that area may have a strange legacy, it may  be related to pypy or  oejskit

Reply:
The check was added in d68c65b493714cc9285e92f057f7ba21eae6c5e3 which added the tuple option -- since the tuple option was removed, I think the check can go too. AFAIU `colitem` is a `Node` and all `Node`s have a `teardown` method defined.

Reply:
Marking as resolved - let me know if you still have concerns.